federation/pkg/kubefed/init/init.go

```diff
@@ -167,11 +171,19 @@ func initFederation(cmdOut io.Writer, config util.AdminConfig, cmd *cobra.Comman
 	}
 
 	// 2. Expose a network endpoint for the federation API server
-	svc, err := createService(hostClientset, initFlags.FederationSystemNamespace, serverName, dryRun)
+	svc, err := createService(hostClientset, initFlags.FederationSystemNamespace, serverName, dryRun, cloudEnvironment)
 	if err != nil {
 		return err
 	}
-	ips, hostnames, err := waitForLoadBalancerAddress(hostClientset, svc, dryRun)
+	ips := []string{}
+	hostnames := []string{}
+	if !dryRun {
+		if cloudEnvironment {
+			ips, hostnames, err = util.WaitForLoadBalancerAddress(hostClientset, svc)
+		} else {
+			ips, err = util.GetFirstNodeIP(hostClientset)
```

Comment:
IMO it falls under the umbrella of this change and isn't too much more work on top of what you already have.  I don't feel too strongly about doing to in this PR vs a different one, although if multiple flags is to be used (later comment) then is makes sense to all this flag with those.